CRM/Report/Form/Contribute/Detail.php

```diff
@@ -588,7 +588,7 @@ public function postProcess() {
     $select = str_ireplace('contribution_civireport.total_amount', 'contribution_soft_civireport.amount', $this->_select);
     $select = str_ireplace("'Contribution' as", "'Soft Credit' as", $select);
     // We really don't want to join soft credit in if not required.
-    if (!empty($this->_groupBy) && !$this->noDisplayContributionOrSoftColumn) {
+    if ((!empty($this->_groupBy) && !$this->noDisplayContributionOrSoftColumn) || (!empty($this->_groupBy) && CRM_Utils_SQL::supportsFullGroupBy())) {
```

Comment:
Or how about 
``` if (!empty($this->_groupBy) && (!$this->noDisplayContributionOrSoftColumn || CRM_Utils_SQL::supportsFullGroupBy()) { ```

Reply:
Per my comments below whether the mysql version is trying to enforce full group by to enforce reliable mysql or not should not be used in the if. We are basically saying 

```
if($thisSiteWantsAccurateData) { 
writeReliableSql();
 } 
else { 
wingit();
}

```
Something like this would make more sense.
```
 if ((!empty($this->_groupBy) && $this->isTableSelected('contribution_soft_civireport')) 
```

But I feel really uncomfortable merging this without a unit test and without a unit test. I'm also deeply uncomfortable with the whole postProcess function of that report & the way it is setting groupBy outside the groupBy function - a unit test covering the contribution soft stuff would get me a lot closer!